Functions/Solution.cs

```diff
@@ -15,74 +16,192 @@ internal class Solution
+            int number = BouncingNumbers.FindNumThatHasMostJumping();
+            Console.WriteLine($"calculation ended, the number is: {number}");
+            return number;
         }
         #endregion
 
         #region Bonus solutions
         public static int Bonus1()
         {
-            /*
-             * TODO: Write code here
-             */
+            string[] digits= new string[] { "one", "two", "three", "four", "five", "six", "seven", "eight", "nine" };
+            string[] specials= new string[] { "ten", "eleven", "twelve", "thirteen", "fourteen", "fifteen", "sixteen", "seventeen", "eighteen", "nineteen" };
+            string[] tens= new string[] { "twenty", "thirty", "forty", "fifty", "sixty", "seventy", "eighty", "ninety"};
+
+            int numOfLetters = 0;
+
+            foreach (string word in digits)
+            {
+                numOfLetters += word.Length;
+            }
+
+            foreach (string word in specials)
+            {
+                numOfLetters += word.Length;
+            }
+
+            for (int tensIndex = 0; tensIndex < tens.Length; tensIndex++)
+            {
+                numOfLetters += tens[tensIndex].Length;
+                for (int digitsIndex = 0; digitsIndex < digits.Length; digitsIndex++)
+                {
+                    numOfLetters += tens[tensIndex].Length;
+                    numOfLetters += digits[digitsIndex].Length;
+                }
+            }
+            Console.WriteLine($"num of letters between one to 100 is: {numOfLetters}");
+            return numOfLetters;
+
+
+        }
+
+        public static int Bonus2(int[] hightMap)
+        {
+            int index = 0;
+            int sumWaterCubes = 0;
+            int currHight = hightMap[index];
+            int maxHight;
+
+            while (currHight == 0 && index<hightMap.Length)
+            {
+                currHight = hightMap[index];
+                index++;
+
+            }
+            maxHight = currHight;
+
+            while (index < hightMap.Length) {
+
+                currHight = hightMap[index];
+
+                if (findWall(hightMap, index, maxHight) && maxHight != 0)
+                {
+                    if (currHight <= maxHight)
+                    {
+                        sumWaterCubes += maxHight - currHight;
+                        index++;
+
+                    }
+                    else
+                    {
+                        maxHight = currHight;
+                        index++;
+                    }
+                }
+                else
+                {
+                    maxHight--;
+                    if (maxHight == 0)
+                    {
+                        break;
+                    }
+                }
+            }
+
+            return sumWaterCubes;
+
         }
 
-        public static int Bonus2(uint[] hightMap)
+        public static Boolean findWall(int[] hightMap, int startIndex, int wallHight)
```

Comment:
This should be in `Helper` file.

Reply:
I agree, this file was created for all helper functions because at this point you weren't expected to go out of your way to create files for each case. The correct way would be to create your own files with appropriate naming and meaning.
Just to make it clear, the `solution` file indeed isn't better than putting in the `helper` file.